File: src/parser/satisfactory/types/property/generic/int64_property.py
from src.parser.byte.binary_readable import BinaryReadable
from src.parser.byte.byte_writer import ByteWriter
from src.parser.satisfactory.types.structs.guid_info import GUIDInfo
from src.parser.satisfactory.types.property.generic.abstract_base_property import AbstractBaseProperty

def is_int64_property(property: any) -> bool:
    return not isinstance(property, list) and property.type == 'Int64Property'

class Int64Property(AbstractBaseProperty):

    def __init__(self, value: str, ue_type: str = 'Int64Property', guid_info: GUIDInfo = None, index: int = 0):
        super().__init__(type='Int64Property', ue_type=ue_type, guid_info=guid_info, index=index)
        self.value = value

    @staticmethod
    def parse(reader: BinaryReadable, ue_type: str, index: int = 0) -> 'Int64Property':
        guid_info = GUIDInfo.read(reader)
        value = Int64Property.read_value(reader)
        return Int64Property(value, ue_type, guid_info, index)

    @staticmethod
    def read_value(reader: BinaryReadable) -> str:
        return str(reader.read_int64())

    @staticmethod
    def calc_overhead(property: 'Int64Property') -> int:
        return 1

    @staticmethod
    def serialize(writer: ByteWriter, property: 'Int64Property') -> None:
        GUIDInfo.write(writer, property.guid_info)
        Int64Property.serialize_value(writer, property.value)

    @staticmethod
    def serialize_value(writer: ByteWriter, value: str) -> None:
        writer.write_int64(int(value))

File: src/parser/satisfactory/types/property/generic/Int64Property.ts
import { BinaryReadable } from '../../../../byte/binary-readable.interface';
import { ByteWriter } from '../../../../byte/byte-writer.class';
import { GUIDInfo } from '../../structs/GUIDInfo';
import { AbstractBaseProperty } from './AbstractBaseProperty';

export const isInt64Property = (property: any): property is Int64Property => !Array.isArray(property) && property.type === 'Int64Property';

export type Int64Property = AbstractBaseProperty & {
    type: 'Int64Property';
    value: string;
};

export namespace Int64Property {

    export const Parse = (reader: BinaryReadable, ueType: string, index: number = 0): Int64Property => {
        const guidInfo = GUIDInfo.read(reader);
        const value = ReadValue(reader);

        return {
            ...AbstractBaseProperty.Create({ index, ueType, guidInfo, type: '' }),
            type: 'Int64Property',
            value,
        } satisfies Int64Property;
    }

    export const ReadValue = (reader: BinaryReadable): string => {
        return reader.readInt64().toString();
    }

    export const CalcOverhead = (property: Int64Property): number => {
        return 1;
    }

    export const Serialize = (writer: ByteWriter, property: Int64Property): void => {
        GUIDInfo.write(writer, property.guidInfo);
        SerializeValue(writer, property.value);
    }

    export const SerializeValue = (writer: ByteWriter, value: string): void => {
        writer.writeInt64(BigInt(value));
    }
}